src/WebJobs.Script/Workers/Rpc/Configuration/DynamicWorkerConfigurationResolver.cs

```diff
@@ -0,0 +1,306 @@
+        /// </summary>
+        public Dictionary<string, RpcWorkerConfig> GetWorkerConfigs()
+        {
+            var workerRuntime = _resolverOptions.CurrentValue.WorkerRuntime;
+            var workerProbingPaths = _resolverOptions.CurrentValue.ProbingPaths;
+
+            // Search for worker configs in probing paths. Returns a dictionary of { FUNCTIONS_WORKER_RUNTIME : RpcWorkerConfig }
+            var runtimeToConfigMap = ResolveWorkerConfigsFromProbingPaths(workerProbingPaths, workerRuntime);
+
+            if (!_resolverOptions.CurrentValue.IsMultiLanguageWorkerEnvironment &&
+                !_resolverOptions.CurrentValue.IsPlaceholderModeEnabled &&
+                !string.IsNullOrWhiteSpace(workerRuntime) &&
+                runtimeToConfigMap.ContainsKey(workerRuntime))
+            {
+                return runtimeToConfigMap;
+            }
+
+            // Search in fallback path if worker cannot be found in probing paths
+            runtimeToConfigMap = ResolveWorkerConfigsFromWithinHost(runtimeToConfigMap);
+
+            return runtimeToConfigMap;
+        }
+
+        /// <summary>
+        /// Resolves worker configurations from the specified probing paths.
+        /// </summary>
+        private Dictionary<string, RpcWorkerConfig> ResolveWorkerConfigsFromProbingPaths(IReadOnlyList<string> workerProbingPaths, string workerRuntime)
+        {
+            var runtimeToConfigPathMap = new Dictionary<string, RpcWorkerConfig>(StringComparer.OrdinalIgnoreCase);
+
+            try
+            {
+                _logger.WorkerProbingPaths(string.Join(", ", workerProbingPaths));
+
+                // Probing path directory structure is: "<probingPath>/<workerRuntimeDir>/<workerVersion>/worker.config.json"
+                foreach (var probingPath in workerProbingPaths)
+                {
+                    if (string.IsNullOrWhiteSpace(probingPath))
+                    {
+                        continue;
+                    }
+
+                    if (!_fileSystem.Directory.Exists(probingPath))
+                    {
+                        _logger.LogDebug("Worker probing path directory does not exist: {probingPath}.", probingPath);
+                        continue;
+                    }
+
+                    foreach (var workerRuntimePath in _fileSystem.Directory.EnumerateDirectories(probingPath))
+                    {
+                        string workerRuntimeDir = Path.GetFileName(workerRuntimePath);
+
+                        // If probing paths are malformed and have duplicate directories of the same language worker (eg. due to different casing)
+                        if (runtimeToConfigPathMap.ContainsKey(workerRuntimeDir))
+                        {
+                            _logger.LogDebug("Skipping duplicate worker runtime directory '{workerRuntimeDir}' in probing path '{probingPath}'.", workerRuntimeDir, probingPath);
```

Comment:
I wonder this should be a hard error. Linux allows folders with same name, but different casing.